loadtest/loadtest.go

```diff
@@ -116,17 +134,20 @@ func (lt *LoadTester) Run() error {
 	lt.mut.Lock()
 	defer lt.mut.Unlock()
 
-	if lt.state != StateStopped {
+	if lt.status.State != StateStopped {
 		return ErrNotStopped
 	}
-	lt.state = StateStarting
+	lt.status.State = StateStarting
 	go lt.handleStatus()
 	for i := 0; i < lt.config.UsersConfiguration.InitialActiveUsers; i++ {
 		if err := lt.addUser(); err != nil {
 			mlog.Error(err.Error())
 		}
 	}
-	lt.state = StateRunning
+	lt.startTimeOnce.Do(func() {
+		lt.status.StartTime = time.Now()
+	})
```

Comment:
Yes, that's what I commented in the JIRA ticket. I wanted to give the idea that the loadtest is only meant to be used once. Otherwise, we would need to have metrics for each session of a test like I mentioned in the ticket.

Maybe I should remove the sync.Once and let the startTime just reset itself if Run is called again ?

Reply:
Totally missed your JIRA comments. The concept of sessions is interesting but as you said I'd wait a bit before complicating things. As I see it we have a few options:

- Leave everything as it is and always increase the stats, support resumption of the same test. If we decide to go with this option I would add the aforementioned value to the stats.
- Completely remove the ability to run a load-test twice which means removing the `sync.Once` call here and also check and return error for `StartTime != 0` in `Run()`. With this option, a new `LoadTester` needs to be created every time we want to run a new load-test.
- Similar to the above, we allow multiple Start/Stop but we reset everything on Stop (time and stats). This avoids creating a new LoadTester.

Reply:
I like the 3rd option. But I think we should reset everything on Run, and only `NumUsers` on Stop. Because we would still want to serve the existing status after a load test has stopped.

Reply:
Yes, most of the reset should really happen on Run. Let's go with that option then.